thunder/clang/__init__.py

```diff
@@ -272,9 +272,6 @@ def convolution(
 def full(
     shape: Sequence[int], fill_value: NumberLike, *, device: DeviceLike, dtype: None | dtypes.dtype = None
 ) -> TensorLike:
-    # Infers dtype from the fill_value when not explicitly provided
```

Comment:
Doesn't the prim require the dtype be specified? So what happens if `dtype` is None here?

@kshitij12345 (and at @IvanYashchuk who may have some thoughts), is the idea of the default dtype torch only, or is it a core language concept? 

If the default dtype is a core language concept, then I would update the core language operations to utilize it. 

If the default dtype is a torch language only concept, then I would make the clang operations require the dtype explicitly (like the primitives) or have them perform their own kind of type inference.

It seems like it'd be easier to have the default dtype be a system-wide concept, that would include the core language operations?

Reply:
The idea of default dtype currently is torch only. The reason for that was, I believe that core lang can be used by other languages eg. numpy and it seems incorrect that details from PyTorch leak into it.

Though I must admit that I am still unclear on what semantics we have/want for the core language (and if it gets a default dtype, should it be same as pytorch?) Though as of now, since PyTorch is the main (and probably only supported higher language), it should probably be ok.

Reply:
`prims` should be strict and require all arguments to be provided with no default value. This pull request is about changing the `torch.full` behavior so I suggest reverting changes made to `clang.full` in this pull request and keep it as is until we start investing more in alternative languages for Thunder.